### PR TITLE
ec_host_cmd: clean-up

### PR DESCRIPTION
**Pull request by niedzwiecki-dawid**
Update configs naming and create a chosen string per backend.

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

**Comment by keith-zephyr** [comment]
Please add a description for the PR as the first comment.

File: doc/services/device_mgmt/ec_host_cmd.rst [hunk 1]
```diff
@@ -45,8 +45,9 @@ The supported backend and peripheral drivers:
 Initialization
 **************
 
-If the application configures the ``zephyr,host-cmd-backend`` chosen node, then the backend
-automatically initializes the host command subsystem by calling :c:func:`ec_host_cmd_init`.
+If the application configures the ``zephyr,host-cmd-espi-backend``/``zephyr,host-cmd-shi-backend``/
+``zephyr,host-cmd-uart-backend`` chosen node, then the backend automatically initializes the host command
+subsystem by calling :c:func:`ec_host_cmd_init`.
 
 If ``zephyr,host-cmd-backend`` is not chosen, the :c:func:`ec_host_cmd_init` function should be
```
Comment (keith-zephyr) [inline doc/services/device_mgmt/ec_host_cmd.rst:52]: Suggestion
"If no backend chosen node is configured, the application must call the `ec_host_cmd_init` function directly.  This way of..."

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: doc/services/device_mgmt/ec_host_cmd.rst [hunk 2]
```diff
@@ -45,8 +45,9 @@ The supported backend and peripheral drivers:
 Initialization
 **************
 
-If the application configures the ``zephyr,host-cmd-backend`` chosen node, then the backend
-automatically initializes the host command subsystem by calling :c:func:`ec_host_cmd_init`.
+If the application configures the ``zephyr,host-cmd-espi-backend``/``zephyr,host-cmd-shi-backend``/
+``zephyr,host-cmd-uart-backend`` chosen node, then the backend automatically initializes the host command
+subsystem by calling :c:func:`ec_host_cmd_init`.
```
[the remark this answers is not in the document]
Comment (semihalf-niedzwiecki-dawid) [reply doc/services/device_mgmt/ec_host_cmd.rst]: Done.

File: subsys/mgmt/ec_host_cmd/backends/Kconfig [hunk 3]
```diff
@@ -3,6 +3,10 @@
 # Copyright (c) 2020 Google LLC
 # SPDX-License-Identifier: Apache-2.0
 
+DT_CHOSEN_ESPI_BACKEND := zephyr,host-cmd-espi-backend
+DT_CHOSEN_SHI_BACKEND := zephyr,host-cmd-shi-backend
+DT_CHOSEN_UART_BACKEND := zephyr,host-cmd-uart-backend
```
[the remark this answers is not in the document]
Comment (semihalf-niedzwiecki-dawid) [reply subsys/mgmt/ec_host_cmd/backends/Kconfig:8]: Technically, it is possible to enable more than one backend at the same time, but I don't think it is our case. We need to have possibility to build in support for a few backend into one image and initialize a proper one in runtime, but it is a different case. So I've added the assert you suggested.

**Review by keith-zephyr — APPROVED** [review]

**Review by GRobertZieba — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**